lib/resty/dns/resolver.lua

```diff
@@ -764,41 +764,31 @@ function _M.query(self, qname, opts)
         local sock = pick_sock(self, socks)
 
         local ok, err = sock:send(query)
-        if not ok then
-            local server = _get_cur_server(self)
-            return nil, "failed to send request to UDP server "
-                .. concat(server, ":") .. ": " .. err
-        end
-
-        local buf, err
-
-        for j = 1, 128 do
-            buf, err = sock:receive(4096)
+        if ok then
+            local buf
 
-            if err then
-                break
-            end
+            for j = 1, 128 do
+                buf, err = sock:receive(4096)
+                if err then
+                    break
+                end
 
-            if buf then
-                local answers
-                answers, err = parse_response(buf, id)
-                if not answers then
-                    if err == "truncated" then
-                        return _tcp_query(self, query, id)
+                if buf then
+                    local answers
+                    answers, err = parse_response(buf, id)
+                    if not answers then
+                        if err == "truncated" then
+                            answers, err =  _tcp_query(self, query, id)
```

Comment:
This `err` assignment never gets used; but this would raise errors if we ever check this project with a static linter like Luacheck. Should we consider either:
- Keeping a record of such errors so in case all retries fail, the user gets a history of the failed attempts? This is also valid for any failed UDP attempt from above ^
- Or dropping this assignment altogether?

(also, there is 2 spaces after the `=` sign)